### PR TITLE
appimageTools: use version

### PR DESCRIPTION
**Pull request by Stunkymonkey**
part of #336576 and #103997

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/build-support/appimage/default.nix [hunk 1]
```diff
@@ -26,7 +26,7 @@ rec {
     ];
   };
 
-  extract = args@{ name ? "${args.pname}-${args.version}", postExtract ? "", src, ... }: pkgs.runCommand "${name}-extracted" {
+  extract = args@{ pname, version , postExtract ? "", src, ... }: pkgs.runCommand "${pname}-${version}-extracted" {
```
Comment (Stunkymonkey) [inline pkgs/build-support/appimage/default.nix:29]: is this fine to do? not sure

File: pkgs/by-name/st/steam-rom-manager/package.nix [hunk 2]
```diff
@@ -10,11 +10,11 @@ appimageTools.wrapType2 rec {
...
   extraInstallCommands = let
-    appimageContents = appimageTools.extract { inherit name src; };
+    appimageContents = appimageTools.extract { inherit pname version src; };
     in ''
-      install -m 444 -D ${appimageContents}/${name}.desktop -t $out/share/applications
-      substituteInPlace $out/share/applications/${name}.desktop \
-        --replace 'Exec=AppRun' 'Exec=${name}'
+      install -m 444 -D ${appimageContents}/${pname}.desktop -t $out/share/applications
+      substituteInPlace $out/share/applications/${pname}.desktop \
+        --replace 'Exec=AppRun' 'Exec=${pname}'
```
Comment (SuperSandro2000) [inline pkgs/by-name/st/steam-rom-manager/package.nix:17]:
```suggestion
      install -m 444 -D ${appimageContents}/steam-rom-manager.desktop -t $out/share/applications
      substituteInPlace $out/share/applications/steam-rom-manager.desktop \
        --replace 'Exec=AppRun' 'Exec=steam-rom-manager'
```

File: pkgs/build-support/appimage/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/build-support/appimage/default.nix:29]: Maybe we should abort if name is passed with an appropriate error?

Comment (Stunkymonkey) [reply pkgs/build-support/appimage/default.nix]: good idea. please look at it again.

File: pkgs/by-name/st/steam-rom-manager/package.nix (hunk 2 above)
Comment (Stunkymonkey) [reply pkgs/by-name/st/steam-rom-manager/package.nix]: applied

File: pkgs/by-name/sa/saleae-logic-2/package.nix [hunk 3]
```diff
@@ -1,14 +1,14 @@
...
   version = "2.4.13";
   src = fetchurl {
     url = "https://downloads.saleae.com/logic2/Logic-${version}-linux-x64.AppImage";
     hash = "sha256-0GIZQKQDY3arDUlxjQKWOHDB3j76xVwkx5H+8q+d0Rc=";
   };
   desktopItem = makeDesktopItem {
-    inherit name;
-    exec = name;
+    name = pname;
+    exec = pname;
```
Comment (Pandapip1) [inline pkgs/by-name/sa/saleae-logic-2/package.nix:11]: Not blocking, but this feels wrong.

**Review by Pandapip1 — APPROVED** [review]

File: pkgs/build-support/appimage/default.nix [hunk 4]
```diff
@@ -26,7 +26,11 @@ rec {
     ];
   };
 
-  extract = args@{ name ? "${args.pname}-${args.version}", postExtract ? "", src, ... }: pkgs.runCommand "${name}-extracted" {
+  extract = args@{ pname, version, name ? null, postExtract ? "", src, ... }:
+  if name != null then
+    throw "The 'name' argument is deprecated. Use 'pname' and 'version' instead to construct the name."
+  else
```
Comment (philiptaron) [inline pkgs/build-support/appimage/default.nix:32]: golf: this is 

```
assert lib.assertMsg (name == null) "The `name` argument is deprecated. Use `pname` and `version` instead to construct the name.";
```

File: pkgs/by-name/de/deskreen/package.nix [hunk 5]
```diff
@@ -4,7 +4,7 @@
 , appimageTools
 }:
 
-stdenvNoCC.mkDerivation (finalAttrs: {
+stdenvNoCC.mkDerivation (finalAttrs: rec {
```
Comment (philiptaron) [inline pkgs/by-name/de/deskreen/package.nix:7]: no `rec` please

File: pkgs/by-name/de/deskreen/package.nix [hunk 6]
```diff
@@ -13,7 +13,7 @@ stdenvNoCC.mkDerivation (finalAttrs: {
     hash = "sha256-0jI/mbXaXanY6ay2zn+dPWGvsqWRcF8aYHRvfGVsObE=";
   };
   deskreenUnwrapped = appimageTools.wrapType2 {
-    name = "deskreen";
+    inherit pname version;
```
Comment (philiptaron) [inline pkgs/by-name/de/deskreen/package.nix:16]:
```suggestion
    inherit (finalAttrs) pname version;
```

File: pkgs/by-name/sa/saleae-logic-2/package.nix (hunk 3 above)
Comment (philiptaron) [reply pkgs/by-name/sa/saleae-logic-2/package.nix:11]: Yeah, let's not do this. https://github.com/NixOS/nixpkgs/issues/277994

File: pkgs/by-name/st/steam-rom-manager/package.nix [hunk 7]
```diff
@@ -10,11 +10,11 @@ appimageTools.wrapType2 rec {
   };
 
   extraInstallCommands = let
-    appimageContents = appimageTools.extract { inherit name src; };
+    appimageContents = appimageTools.extract { inherit pname version src; };
     in ''
-      install -m 444 -D ${appimageContents}/${name}.desktop -t $out/share/applications
-      substituteInPlace $out/share/applications/${name}.desktop \
-        --replace 'Exec=AppRun' 'Exec=${name}'
+      install -m 444 -D ${appimageContents}/steam-rom-manager.desktop -t $out/share/applications
```
Comment (philiptaron) [inline pkgs/by-name/st/steam-rom-manager/package.nix:15]: yes, this duplication is the right way to go

**Review by philiptaron — CHANGES_REQUESTED** [review]
Requesting changes in the two spots noted. I also golfed the implementation a bit; take that if you want or not, no strong feelings.

File: pkgs/by-name/st/steam-rom-manager/package.nix (hunk 7 above)
Comment (Stunkymonkey) [reply pkgs/by-name/st/steam-rom-manager/package.nix:15]: I assume this is a compliment. if so, please mark it as resolved.

**Review by philiptaron — APPROVED** [review]

**Comment by philiptaron** [comment]
I haven't built it in the big machine yet.

**Comment by philiptaron** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 359930`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 6 packages built:</summary>
  <ul>
    <li>cables</li>
    <li>deskreen</li>
    <li>quba</li>
    <li>saleae-logic-2</li>
    <li>ssb-patchwork</li>
    <li>steam-rom-manager</li>
  </ul>
</details>

**Pull request merged**